doc/contrib/release.rst

```diff
@@ -0,0 +1,13 @@
+.. _release:
+
+XGBoost Release Policy
+=======================
+
+Versioning Policy
+---------------------------
+
+Starting from XGBoost 1.0.0, each XGBoost release will be versioned as [MAJOR].[FEATURE].[MAINTENANCE]
+
+* MAJOR: We gurantee the API compatibility across releases with the same major version number. We expect to have a 1+ years development period for a new MAJOR release version.
```

Comment:
I like this document and I like that xgboost will guarantee API compatibility.
I have only one comment:

Should we specify that we guarantee binary compatibility (as a oppose to source code one) for C++ and Java parts?
Pros of a binary compatibility that one could just update a shared library or xgboost jar without rebuilding the application which uses it.

Cons. One should be extra careful on some changes which are compatible on source code level but not binary level. Example is adding a new parameter to a public API method with a default value.

Reply:
it's a great suggestion

MIMA looks like to be a great tool to detect jvm-level issues

for C++, https://lvc.github.io/abi-compliance-checker/ looks like to be fitting here...but I would raise this question to @RAMitchell @hcho3 and @trivialfis , how is it important for c++ library of xgb to guarantee binary compatibility since we are not mostly wrap c++ library with other language bindings?

Reply:
@CodingCat We can start by guaranteeing the binary compatibility for the C API functions first, since all language bindings use C API, not C++ API.

Reply:
@CodingCat I think you are right that we do not need to guarantee binary compatibility for native code of xgboost. But the answer to this depends whether there is someone who is using xgboost as a shared library

Reply:
@trams per @hcho3, we are not going to support such scenario, and I agree with him considering the cost